web/pdf_page_view.js

```diff
@@ -90,125 +78,143 @@ var PDFPageView = (function PDFPageViewClosure() {
+    this.paintedViewportMap.delete(zoomLayerCanvas);
+    // Zeroing the width and height causes Firefox to release graphics
+    // resources immediately, which can greatly reduce memory consumption.
+    zoomLayerCanvas.width = 0;
+    zoomLayerCanvas.height = 0;
+
+    if (removeFromDOM) {
+      // Note: `ChildNode.remove` doesn't throw if the parent node is undefined.
+      this.zoomLayer.remove();
+    }
+    this.zoomLayer = null;
+  }
 
-      if (!currentZoomLayerNode) {
-        if (this.canvas) {
-          this.paintedViewportMap.delete(this.canvas);
-          // Zeroing the width and height causes Firefox to release graphics
-          // resources immediately, which can greatly reduce memory consumption.
-          this.canvas.width = 0;
-          this.canvas.height = 0;
-          delete this.canvas;
-        }
-        this._resetZoomLayer();
+  reset(keepZoomLayer = false, keepAnnotations = false) {
+    this.cancelRendering();
+
+    let div = this.div;
+    div.style.width = Math.floor(this.viewport.width) + 'px';
+    div.style.height = Math.floor(this.viewport.height) + 'px';
+
+    let childNodes = div.childNodes;
+    let currentZoomLayerNode = (keepZoomLayer && this.zoomLayer) || null;
+    let currentAnnotationNode = (keepAnnotations && this.annotationLayer &&
+                                 this.annotationLayer.div) || null;
+    for (let i = childNodes.length - 1; i >= 0; i--) {
+      let node = childNodes[i];
+      if (currentZoomLayerNode === node || currentAnnotationNode === node) {
+        continue;
       }
-      if (this.svg) {
-        this.paintedViewportMap.delete(this.svg);
-        delete this.svg;
+      div.removeChild(node);
+    }
+    div.removeAttribute('data-loaded');
+
+    if (currentAnnotationNode) {
+      // Hide the annotation layer until all elements are resized
+      // so they are not displayed on the already resized page.
+      this.annotationLayer.hide();
+    } else {
+      this.annotationLayer = null;
+    }
+
+    if (!currentZoomLayerNode) {
+      if (this.canvas) {
+        this.paintedViewportMap.delete(this.canvas);
+        // Zeroing the width and height causes Firefox to release graphics
+        // resources immediately, which can greatly reduce memory consumption.
+        this.canvas.width = 0;
+        this.canvas.height = 0;
+        delete this.canvas;
       }
+      this._resetZoomLayer();
+    }
+    if (this.svg) {
+      this.paintedViewportMap.delete(this.svg);
+      delete this.svg;
+    }
+
+    this.loadingIconDiv = document.createElement('div');
+    this.loadingIconDiv.className = 'loadingIcon';
+    div.appendChild(this.loadingIconDiv);
+  }
 
-      this.loadingIconDiv = document.createElement('div');
-      this.loadingIconDiv.className = 'loadingIcon';
-      div.appendChild(this.loadingIconDiv);
-    },
+  update(scale, rotation) {
+    this.scale = scale || this.scale;
+    this.rotation = rotation || this.rotation;
```

Comment:
This isn't correct, since it will effectively prevent setting `this.rotation` to zero, if the rotation has previously been set to a non-zero value; please revert this change!